src/components/NewPaymentRequest.js

```diff
@@ -67,7 +67,7 @@ class NewPaymentRequest extends React.Component {
   }
 
   componentWillUnmount() {
-    this.willFocusEvent.remove();
+    this.focusEvent();
   }
```

Comment:
Agreed! By upgrading to functional components we would also be able to use the more efficient [`useFocusEffect` hook](https://reactnavigation.org/docs/5.x/use-focus-effect) instead of creating and managing event listeners.